spring-pulsar/src/test/java/org/springframework/pulsar/core/DefaultExpressionResolverTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2012-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.config.BeanExpressionContext;
+import org.springframework.beans.factory.config.BeanExpressionResolver;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+
+/**
+ * Unit tests for {@link DefaultExpressionResolver}.
+ *
+ * @author Jonas Geiregat
+ */
+class DefaultExpressionResolverTest {
```

Comment:
Nice test @jonasgeiregat! I think it would be a good idea to a couple of tests to DTR tests as well.

### Case 1
**Does my expression resolver get called?**

This would fit nicely in the existing nested "annotated" test class w/in DTRT, something like:

```java
private static final String bazTopicExpression = "#{someExpression}";

@Test
void annotatedMessageTypeWithTopicExpressionIsResolved() {
	var mockExpressionResolver = mock(ExpressionResolver.class);
	when(mockExpressionResolver.resolveToString(bazTopicExpression)).thenReturn(Resolved.of(bazTopic));
	var expressionTopicResolver = new DefaultTopicResolver(mockExpressionResolver);
	assertThat(expressionTopicResolver.resolveTopic(null, BazWithTopicExpression.class, 
			() -> defaultTopic).value().orElse(null))
			.isEqualTo(bazTopic);
	verify(mockExpressionResolver, times(1)).resolveToString(bazTopicExpression);
}

@PulsarMessage(topic = bazTopicExpression)
record BazWithTopicExpression(String value) {
}

```

### Case 2
**Does my expression resolver get derived during call of BFA?**

We could easily put this in the same nested "annotated" test class within DTRT and just mock the BF like you are doing in this test.  However, because:

- we know that this will likely go away in the future once SB passes in via AC. 
- it would probably be a good thing to also put 1-2 sanity resolver tests in the existing `Reactive/PulsarListener/Reader` ITs which will call BFA and this will in turn indirectly cover this "Does my resolver get created from BF during BFA?". This way when/if the BFA is removed, all cases are still covered (i.e this test would still be valid even w/o the BFA impl.

Wdyt? 

Also, we appreciated all of your contributions and time you are putting into this and if you don't have time for another round on this I would gladly run w/ it as you have left it in beautiful shape. But also, there is no real rush and these changes can wait until you have time to get back to it - totally up to you. Thanks again. 🙇🏻‍♂️

Reply:
No should thank you for your patience and valuable feedback, I'm learning tons. I updated everything until now just need to take a look at the ITs as they are new to me but think I'll manage.